lib/menu.c

```diff
@@ -954,4 +1103,207 @@ bm_menu_run_with_key(struct bm_menu *menu, enum bm_key key, uint32_t unicode)
+
+    uint32_t displayed = 0;
+    if (menu->renderer->api.get_displayed_count)
+        displayed = menu->renderer->api.get_displayed_count(menu);
+
+    if (!displayed)
+        displayed = count;
+
+    if (!menu->lines) {
+        if (pointer.axes[BM_POINTER_AXIS_VERTICAL].valid) {
+            if (0 < pointer.axes[BM_POINTER_AXIS_VERTICAL].value) {
+                menu_next(menu, count, menu->wrap);
+            } else {
+                menu_prev(menu, count, menu->wrap);
+            }
+        }
+        if (pointer.event_mask & POINTER_EVENT_BUTTON && pointer.state == POINTER_STATE_PRESSED) {
+            switch (pointer.button) {
+                case BM_POINTER_KEY_PRIMARY:
+                    {
+                        struct bm_item *highlighted = bm_menu_get_highlighted_item(menu);
+                        if (highlighted && !bm_menu_item_is_selected(menu, highlighted))
+                            list_add_item(&menu->selection, highlighted);
+                    }
+                    return BM_RUN_RESULT_SELECTED;
+            }
+        }
+       return BM_RUN_RESULT_RUNNING;
+    }
+
+    if (pointer.axes[BM_POINTER_AXIS_VERTICAL].valid) {
+        if (0 < pointer.axes[BM_POINTER_AXIS_VERTICAL].value) {
+            menu_scroll_down(menu, count);
+        } else {
+            menu_scroll_up(menu, count);
+        }
+    }
+
+    if (pointer.event_mask & POINTER_EVENT_MOTION) {
+        menu_point_select(menu, pointer.pos_x, pointer.pos_y, displayed);
+    }
+
+    if (pointer.event_mask & POINTER_EVENT_BUTTON && pointer.state == POINTER_STATE_PRESSED) {
+        switch (pointer.button) {
+            case BM_POINTER_KEY_PRIMARY:
+                {
+                    struct bm_item *highlighted = bm_menu_get_highlighted_item(menu);
+                    if (highlighted && !bm_menu_item_is_selected(menu, highlighted))
+                        list_add_item(&menu->selection, highlighted);
+                }
+                return BM_RUN_RESULT_SELECTED;
+        }
+    }
+
+    return BM_RUN_RESULT_RUNNING;
+}
+
+enum bm_run_result
+bm_menu_run_with_touch(struct bm_menu *menu, struct bm_touch touch, uint32_t unicode)
+{
+    (void) unicode;
+    uint32_t count;
+    bm_menu_get_filtered_items(menu, &count);
+
+    uint32_t displayed = 0;
+    if (menu->renderer->api.get_displayed_count)
+        displayed = menu->renderer->api.get_displayed_count(menu);
+
+    if (!displayed)
+        displayed = count;
+
+    if (!menu->lines) {
+        // Not implemented yet
+        return BM_RUN_RESULT_RUNNING;
+    }
+
+    const size_t nmemb = sizeof(touch.points) / sizeof(struct bm_touch_point);
+
+    uint16_t count_down = 0;
+    for (size_t i = 0; i < nmemb; ++i) {
+        struct bm_touch_point point = touch.points[i];
+        if (point.event_mask & TOUCH_EVENT_DOWN) {
+            count_down += 1;
+        }
+    }
+
+    if (count_down == 2) {
+        int16_t scroll_count = 0;
+        int16_t scroll_directions[nmemb];
```

Comment:
can we make this not use VLA? I assume `nmemb` won't ever be more than 5 so we can hardcode the scroll_directions to [5]?

Reply:
I think this code came from a long time. Now I enforce 2 in the backend wayland so lets just enforce 2 everywhere. It would be easy to change anyway.